pkg/adapter/common/backoff_test.go

```diff
@@ -127,47 +127,90 @@ func TestRun(t *testing.T) {
 			waitReturn: true,
 			err:        assert.AnError,
 		},
-		"fn resets backoff duration": {
-			fn: func(ctx context.Context) (bool, error) {
-				counter++
-				if counter > 3 {
-					if time.Since(start).Round(time.Second) != 3*time.Second {
```

Comment:
This time-based expectation was very optimistic, considering `start` is shared between all tests, and delaying one previous test could have  made this one fail.

I rewrote it using a channel to control the number of executions of the function from the main test logic.

Reply:
nice, I remember I was having a hard time with this test and the time-based check was a sort of desperate decision